src/omp/backend.f90

```diff
@@ -147,10 +150,36 @@ subroutine transeq_z_omp(self, du, dv, dw, u, v, w, dirps)
       type(dirps_t), intent(in) :: dirps
 
       ! u, v, w is reordered so that we pass w, u, v
-      !call self%transeq_omp_dist(dw, du, dv, w, u, v, dirps)
+      call self%transeq_omp_dist(dw, du, dv, w, u, v, dirps)
 
    end subroutine transeq_z_omp
 
+   subroutine transeq_omp_dist(self, du, dv, dw, u, v, w, dirps)
+      implicit none
+
+      class(omp_backend_t) :: self
+      class(field_t), intent(inout) :: du, dv, dw
+      class(field_t), intent(in) :: u, v, w
+      type(dirps_t), intent(in) :: dirps
+      class(field_t), pointer :: duu, d2u, uu, du_temp
+
+      ! du
+      du_temp => self%allocator%get_block()
+      call tds_solve_omp(self, du_temp, u, dirps, dirps%der1st)
+
+      duu => self%allocator%get_block()
+      uu => self%allocator%get_block()
+      call vecmul_omp(uu, u, u, dirps)
+      call tds_solve_omp(self, duu, uu, dirps, dirps%der1st_sym)
+
+      d2u => self%allocator%get_block()
+      call tds_solve_omp(self, d2u, u, dirps, dirps%der2nd)
+
+
+
+
+   end subroutine transeq_omp_dist
+
```

Comment:
This is not the way we want to implement transeq stuff on the OpenMP backend. We can't take advantage of the cache and reduce the data movements if we implement in this way.

We should instead implement a subroutine that is similar to exec_dist_tds_compact.
https://github.com/xcompact3d/x3d2/blob/main/src/omp/exec_dist.f90#L14C15-L14C36

And the idea is we call the der_univ_dist to get du/dx, duu/dx, and d2u/dx2 one after another as we loop through batches of lines we have in the domain. (So a loop over the 3rd dimension of the arrays we have, aka the group number.

It should look like the loop below, but with 3 distinct calls to der_univ_dist inside the loop with correct set of tdsops instances.
https://github.com/xcompact3d/x3d2/blob/main/src/omp/exec_dist.f90#L40-L47

This will also make an operation like vecmul_omp unnecessary, saving some data movement.

Reply:
ah yes, makes sense indeed. I will do that